File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        int a=0,maxi;
        int mini=prices[0];
        if(n==1)return 0;
        for(int i=1;i<n;i++){
            mini=min(mini,prices[i]);
            a=prices[i] - mini;
            maxi=max(maxi,a);

        }
        if(maxi>0){
            return maxi;
        }
       return 0; 
    }
    
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxi = 0;
        int maxProfit =0;
        int min= prices[0];
        int n = prices.length;
        if(n ==0)return 0;//as if single element so how to buy and sell
        for(int i=1;i<n;i++){
            min = Math.min(min, prices[i]);
            if(prices[i] - min >=0){
                maxi = Math.max(maxi, prices[i] - min);
            }
        }
        return maxi;
    }
}